aten/src/ATen/native/vulkan/Vulkan.cpp

```diff
@@ -896,6 +908,79 @@ VBuffer makeUniformConstBuffer(void* ptr, VkDeviceSize size) {
   return constBuffer;
 }
 
+ComputeUnitFactory::ComputeUnitFactory(const VkDevice& device)
+    : device_(device) {
+  VkPipelineCacheCreateInfo createInfo{};
+  createInfo.sType = VK_STRUCTURE_TYPE_PIPELINE_CACHE_CREATE_INFO;
+  createInfo.pNext = nullptr;
+  createInfo.flags = 0;
+  createInfo.initialDataSize = 0;
+  createInfo.pInitialData = nullptr;
+  VK_CHECK(vkCreatePipelineCache(
+      device_, &createInfo, nullptr /* allocator */, &pipelineCache_));
+}
+
+ComputeUnitFactory::~ComputeUnitFactory() {
+  vkDestroyPipelineCache(device_, pipelineCache_, nullptr /* allocator */);
+}
+
+std::string ComputeUnitFactory::getCacheKey(
+    const char* key,
+    WorkGroupSize& workGroupSize) {
+  std::stringstream ss;
+  ss << key << ':' << workGroupSize.x << ':' << workGroupSize.y << ':'
+     << workGroupSize.z;
```

Comment:
This kind of looks like an expensive operation to construct a key for lookups if those lookups are frequent but on the other hand it may never end up mattering so I think it's fine to roll with it for now.